🐛 Fix missing active theme breaks design screen

kristophesankar opened the pull request:
refs #15505

When starting Ghost with a missing active theme, the design settings screen and change theme screen both end up in a broken state with the user unable to select a new theme as the active one. 

The design screen has no default (or blank) slate, and so shows a preview of an empty theme.
- First added a new default screen to serve as a placeholder for when the state contains no active theme.
- Added a check for when there was no active theme, then redirects the user to the default screen .

The change theme screen wants to set an active property on the theme that should be active in the theme list.
- Added a check to see whether there is an active theme set. If there isn't one, don't bother trying to add the active property.

kristophesankar committed:
:bug: Added no-theme route to settings/design

refs 15505
- Added the route no-theme as a redirect for when there is no active theme
- Contains a message informing the user of no active theme and a button to select a theme

kristophesankar committed:
:bug: Fixed broken change-theme route

refs 15505
- the store has no theme with the active property set
  - the change-theme controller referenced an undefined variable activeTheme which relied on an active theme
  - a boolean check was added ensuring the value is not undefined
  - this enabled the change-theme route to render properly
- the design route shows a blank page when no theme is active
  - the afterModel hook was added and redirects the user to the no-theme route if there is no currently active theme
- the user is redirected to the design route if there is an active theme

kristophesankar committed:
Fixed formatting issues
refs 15505

- fixed semicolons and indentations

ErisDS commented:
This looks absolutely spot-on to me 👌 but I've requested a second pair of eyes from @kevinansfield on the ember code 🙂

kevinansfield commented on ghost/admin/app/routes/settings/design/index.js:
```diff
@@ -7,9 +7,19 @@ export default class SettingsDesignIndexRoute extends AuthenticatedRoute {
 
     confirmModal = null;
     hasConfirmed = false;
+    themes = this.store.findAll('theme');
```
Setting properties to promises like this in singletons (routes, controllers, services) isn't good practice because there's no control over when the class instance will be instantiated and the promise is kicked off.

If we know that themes are loaded in a higher-level route and already in the store (eg, by adding the `findAll` to the list of [promises that the `routes/settings/design.js` model hook waits for](https://github.com/TryGhost/Ghost/blob/main/ghost/admin/app/routes/settings/design.js#L19-L23)) then we can confidently can use a non-promise-returning method like `themes = this.store.peekAll('theme')` here.

kristophesankar replied:
This makes a lot of sense! I've gone ahead and pushed the changes.

kristophesankar committed:
:bug: Move promise into a higher-level route

closes #15505
- moved the findAll promise that loads the theme to a higher level route
- changed findAll to peekAll because we can guarantee that the theme is loaded now

kristophesankar commented on ghost/admin/app/routes/settings/design/no-theme.js:
```diff
@@ -0,0 +1,16 @@
+export default class SettingsDesignNoThemeRoute extends AuthenticatedRoute {
+    @service store;
+
+    themes = this.store.findAll('theme');
```
Thanks for catching that. I just amended the commit to include that change.

ErisDS commented:
Hi @kristophesankar thanks so much for this PR. It has now been merged 🎉 and will appear in the next release of Ghost - usually Fridays.

I'm not sure if you found this through hacktoberfest, but I've added the `hacktoberfest-accepted` label to this PR to make sure it counts. We have plenty of open issues and ongoing projects for hacktoberfest and beyond. We would love to see you around again 👋
